src/components/GoBackButton.jsx

```diff
@@ -3,8 +3,32 @@ import PropTypes from 'prop-types';
 
 const GoBackButton = ({ history, styleName }) => (
   <div
-    onClick={history.goBack}
-    onKeyPress={history.goBack}
+    onClick={() => {
+      // ths back button will go a lower nested route inside of the DApp
+      // if the browser history is zero
+      //  this should be the default behavior of the back button
+      if (history.length < 1) {
```

Comment:
Could you extract the logic that is repeated in both `onClick` and `onKeyPress` handlers to a function? Also I don't think this solves the issue of accessing giveth from external site. I think the back button should always go one route lower and if the user wants to go back they use the browser back button.

Reply:
yes i agree, I can extract the logic into a utility function, without the conditional statement which isn't necessary